pkg/apis/apps/v1/conversion_test.go

```diff
@@ -0,0 +1,61 @@
+/*
+Copyright 2017 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1_test
+
+import (
+	"testing"
+
+	appsv1 "k8s.io/api/apps/v1"
+	"k8s.io/apimachinery/pkg/util/intstr"
+	"k8s.io/kubernetes/pkg/api"
+	"k8s.io/kubernetes/pkg/apis/extensions"
+
+	apiequality "k8s.io/apimachinery/pkg/api/equality"
+)
+
+func TestV1RollingUpdateDaemonSetConversion(t *testing.T) {
+	intorstr := intstr.FromInt(1)
+	testcases := map[string]struct {
+		rollingUpdateDs1 *extensions.RollingUpdateDaemonSet
+		rollingUpdateDs2 *appsv1.RollingUpdateDaemonSet
+	}{
+		"RollingUpdateDaemonSet Conversion 2": {
+			rollingUpdateDs1: &extensions.RollingUpdateDaemonSet{MaxUnavailable: intorstr},
+			rollingUpdateDs2: &appsv1.RollingUpdateDaemonSet{MaxUnavailable: &intorstr},
+		},
+	}
+
+	for k, tc := range testcases {
+		// extensions -> v1
+		internal1 := &appsv1.RollingUpdateDaemonSet{}
```

Comment:
Do we want to test conversions between v1beta2 or are we confident that transitivity covers this?

Reply:
Conversion only happens between internal and external version, i.e. extensions/ <--> apps/v1, but not apps/v1beta2 <--> apps/v1

Reply:
ack